Dockerfile

```diff
@@ -56,3 +66,86 @@ ENTRYPOINT ["docker-entrypoint.sh"]
 # # By default you'll get a single-node development server that stores everything
 # # in RAM and bootstraps itself. Don't use this configuration for production.
 CMD ["server", "-dev"]
+
+
+## UBI DOCKERFILE ##
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.5 as ubi
+
+ARG BIN_NAME
+# NAME and PRODUCT_VERSION are the name of the software in releases.hashicorp.com
+# and the version to download. Example: NAME=vault PRODUCT_VERSION=1.2.3.
+ARG NAME=vault
+ARG PRODUCT_VERSION
```

Comment:
I don't think we need the `NAME` arg here as I don't see it referenced anywhere. However we should have a `PRODUCT_REVISION` arg as that's passed in automatically by the Action.

```suggestion
# PRODUCT_VERSION is the version built dist/$TARGETOS/$TARGETARCH/$BIN_NAME,
# which we COPY in later. Example: PRODUCT_VERSION=1.2.3.
ARG PRODUCT_VERSION
ARG PRODUCT_REVISION
```

Reply:
Definitely, `PRODUCT_VERSION` and `PRODUCT_REVISION` are both set automatically by the Action when it runs `docker build` so all Dockerfiles and stages within them should have those two ARGs, even if they're not otherwise referenced, as I think the build would fail if an unrecognised build arg is set.